enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+      The kube-controller-manager logs may contain more details.)
+    reason: IngressControllerUnavailable
+    status: "False"
+    type: Available
+
+```
+
+- Are there any pre-requisites for this EP ?
+  Yes, one must [allocate Elastic IP address](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating)
+  in their AWS environment which could be then used in the `eipAllocations` field of the IngressController CR.
+
+- What happens when the cluster-admin specified invalid EIPs in install-config.yaml at installation time (day 0) ?
+  We would try to get the installer feasibly validate the install config for day 0. However, we still need verify this
+  with the installer team. Mostly, we will need to add validations in [pkg/types/aws/validation/platform.go](https://github.com/openshift/installer/blob/master/pkg/types/aws/validation/platform.go)
+
+Note: The following questions are applicable when `auto-delete-load-balancer` annotation is set in the IngressController as it tries to recreate the update Load Balancer.
+If not set then the LoadBalancerProgressing Status will be set to "True" with an message as described in [subsection of Test Plan](#update-an-ingress-controller-eipallocations-by-adding-auto-delete-load-balancer-annotation-in-the-ingress-controller-cr-which-already-has-eipallocations-set)
+
+- What happens when a cluster-admin adds invalid EIPs ?
+
+  The load balancer creation would not proceed and fail as per details mentioned in [subsection of Test Plan](#Test when Ingress Controller CR has been provided with invalid EIPAllocations.)
+
+- What happens when a cluster-admin adds number of EIPs which are not equal to the number of public subnets for the VPC id tagged with cluster name.
+
+  The load balancer creation would not proceed and fail as per details mentioned in [subsection of Test Plan](#test-when-ingress-controller-cr-has-been-provided-with-number-of-eips-not-equal-to-the-number-of-public-subnets-for-vpc-tagged-with-your-cluster-name)
+
+- What happens when a cluster-admin adds nothing under eipAllocations ?
+  
+  The load balancer would be created with no eipAllocations and the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` will be removed.
+  For more check [this](#Cluster admin wants operator to update the Load Balancer service for updated Ingress Controller eipAllocations)
+  The NLB will get the regular IP address assigned in that case.
+
+## Test Plan
+
+### This EP will be covered by unit tests in the following:
+- cluster-ingress-operator
+- installer
+
+### This  EP will also be covered by end to end tests.
```

Comment:
Is there a plan to add an e2e to installer/install time/Day-0?

cc @patrickdillon @r4f4

Reply:
I am not sure how to add one for the installer. I need to check on this.